controllers/adminsController/adminDeleteComment.js

```diff
@@ -0,0 +1,61 @@
+const Comments = require("../../models/comments");
+const Applications = require("../../models/applications");
+const { ObjectId } = require("mongoose").Types;
+
+// Utilities
+const CustomError = require("../../utils/customError");
+const responseHandler = require("../../utils/responseHandler");
+const commentHandler = require("../../utils/commentHandler");
+
+/**
+ * @author David Okanlawon
+ *
+ * Deletes a single comment.
+ *
+ * @param {*} req - The request object
+ * @param {*} res - The response object
+ * @param {*} next - The function executed to call the next middleware
+ */
+const adminDeleteComment = async (req, res, next) => {
+  const commentId = req.body.commentId;
+  const { organizationId } = req.token;
+
+  if (!ObjectId.isValid(organizationId))
+    return next(new CustomError(401, "unauthorised request"));
+
+  try {
+    //find comment
+    const comment = await Comments.findById(commentId);
+    if (!comment) {
+      return next(new CustomError(404, "Comment not found"));
+    }
+
+    if (comment.flags.length === 0) {
+      return next(new CustomError(400, "You can only delete flagged comments"));
+    }
+
+    //get applicaitonId of comment find out if it belongs to organization
+    const application = await Applications.findOne({
+      _id: comment.applicationId,
+      organizationId: organizationId,
+    });
+
+    if (!application) {
+      return next(new CustomError(404, "Invalid comment or comment not found"));
+    }
```

Comment:
I think you should first check if the comment belongs to the organization before you check the comments flags. Because what if the flags are more than one only to be denied access to delete the comment because of an invalid application ID at the next step.

This is just me thinking out loud. The code works perfectly as is but I wonder if the flags check would make more sense after first verifying that we have access to the comment.

Reply:
I thought of that as well but confirming the application takes a second trip to the DB. I was thinking performance. But I'll revert if you think that is a better option after considering this cos the thought nagged in my head as well

Reply:
If checking the length of flags first doesn't require a trip to the DB then I guess the current implementation would be a better option. Even though semantically it makes better sense the other way. Performance is also a factor we should consider. So the current implementation is better.